src/DataWrangling/ECCO/ECCODarwin.jl

```diff
@@ -0,0 +1,485 @@
+               mask = nothing,
+               horizontal_halo = (7, 7),
+               cache_inpainted_data = false)
+
+Return a `Field` on `architecture` described by `ECCOMetadata` with
+`horizontal_halo` size.
+If not `nothing`, the `inpainting` method is used to fill the cells
+within the specified `mask`. `mask` is set to `ECCO_mask` for non-nothing
+`inpainting`.
+"""
+function ECCO_field(metadata::ECCODarwinMetadata;
+                    architecture = CPU(),
+                    inpainting = NearestNeighborInpainting(Inf),
+                    mask = nothing,
+                    horizontal_halo = (7, 7),
+                    cache_inpainted_data = true)
+
+    z = ClimaOcean.DataWrangling.ECCO.ECCO_z
+    input_data_grid = LatitudeLongitudeGrid(architecture; 
+                z,
+                size      = (720, 360, 50),
+                latitude  = (-90, 90),
+                longitude = (0, 360),
+    )
+    field_input_data = Field{Center, Center, Center}(input_data_grid)
+    inpainted_path = inpainted_metadata_path(metadata)
+
+    if !isnothing(inpainting) && isfile(inpainted_path)
+        file = jldopen(inpainted_path, "r")
+        maxiter = file["inpainting_maxiter"]
+
+        # read data if generated with the same inpainting
+        if maxiter == inpainting.maxiter
+            data = file["data"]
+            close(file)
+            copyto!(parent(field_input_data), data)
+            return field_input_data
+        end
+
+        close(file)
+    end
+
+    download_dataset(metadata)
+    path = metadata_path(metadata)
+
+    # Open ECCODarwin binary data file
+    data = ECCODarwinModelData(
+                path, 
+                ECCODarwinModelMeta(
+                    replace(path,"data"=>"meta"),
+                ),
+                ECCODarwinMeshGrid(
+                    metadata.version,
+                ),
+    )
+
+    if variable_is_three_dimensional(metadata)
+        data = reverse(data, dims=3)
+    end        
+    
+    # Convert data from Union(FT, missing} to FT
+    FT = eltype(field_input_data)
+    data[ismissing.(data)] .= 1e10 # Artificially large number!
+    data = if location(field_input_data)[2] == Face # ?
+        new_data = zeros(FT, size(field_input_data))
+        new_data[:, 1:end-1, :] .= data
+        new_data    
+    else
+        data = Array{FT}(data)
+    end
+    
+    # ECCO4 data is on a -180, 180 longitude grid as opposed to ECCO2 data that
+    # is on a 0, 360 longitude grid. To make the data consistent, we shift ECCO4
+    # data by 180 degrees in longitude
+    if metadata.version isa ECCO4DarwinMonthly 
+        Nx = size(data, 1)
+        if variable_is_three_dimensional(metadata)
+            shift = (Nx ÷ 2, 0, 0)
+        else
+            shift = (Nx ÷ 2, 0)
+        end
+        data = circshift(data, shift)
+    end
+    set!(field_input_data, data)
+    fill_halo_regions!(field_input_data)
+
+    if !isnothing(inpainting)
```

Comment:
I think this inpainting should be done on the original LLC data, which is located in the `ECCODarwinModelData` function at `ma.f`. The mask seems to be when elements are equal to zero.